daemon/graphdriver/fsdiff.go

```diff
@@ -11,29 +11,21 @@ import (
 	"github.com/docker/docker/pkg/ioutils"
 )
 
-// naiveDiffDriver takes a ProtoDriver and adds the
-// capability of the Diffing methods which it may or may not
-// support on its own. See the comment on the exported
-// NaiveDiffDriver function below.
-// Notably, the AUFS driver doesn't need to be wrapped like this.
-type naiveDiffDriver struct {
-	ProtoDriver
-}
-
-// NaiveDiffDriver returns a fully functional driver that wraps the
-// given ProtoDriver and adds the capability of the following methods which
+// NaiveDiffDriver takes a ProtoDriver and adds the
+// capability of the following methods which
 // it may or may not support on its own:
 //     Diff(id, parent string) (archive.Archive, error)
 //     Changes(id, parent string) ([]archive.Change, error)
 //     ApplyDiff(id, parent string, diff archive.Reader) (size int64, err error)
 //     DiffSize(id, parent string) (size int64, err error)
-func NaiveDiffDriver(driver ProtoDriver) Driver {
-	return &naiveDiffDriver{ProtoDriver: driver}
+// Notably, the AUFS driver doesn't need to be wrapped like this.
+type NaiveDiffDriver struct {
+	ProtoDriver
```

Comment:
And for that matter, since this will no longer be enforced as a function, then leave a comment that that the inheriting `ProtoDriver` ought to be a pointer.